pkg/nuctl/test/function_test.go

```diff
@@ -681,6 +684,122 @@ func (suite *functionDeployTestSuite) TestBuildAndDeployFromFileWithOverriddenAr
 	suite.Require().Contains(suite.outputBuffer.String(), randomString)
 }
 
+func (suite *functionDeployTestSuite) TestDeployWithResourceVersion() {
+
+	// TODO: when enable some sort of resource validation on other platforms, remove platform ensuring
+	suite.ensureRunningOnPlatform("kube")
+
+	// read and parse the function we're gonna test
+	functionConfig := functionconfig.Config{}
+	uniqueSuffix := "-" + xid.New().String()
+	functionPath := path.Join(suite.GetFunctionsDir(), "common", "json-parser-with-function-config", "python")
+	functionBody, err := ioutil.ReadFile(filepath.Join(functionPath, build.FunctionConfigFileName))
+	suite.Require().NoError(err)
+	err = yaml.Unmarshal(functionBody, &functionConfig)
+	suite.Require().NoError(err)
+
+	// name uniqueness
+	functionConfig.Meta.Name = functionConfig.Meta.Name + uniqueSuffix
+
+	// ensure no resource version
+	functionConfig.Meta.ResourceVersion = ""
+
+	// --- step 1 ---
+	// deploy first time, should successfully create the function
+
+	// write function config to file
+	functionConfigPath, err := suite.writeFunctionConfigToTemp(&functionConfig,
+		"resource-version-*.yaml")
+	suite.Require().NoError(err)
+
+	// remove when done
+	defer os.RemoveAll(functionConfigPath)
+
+	// deploy with temp, expect to pass
+	err = suite.ExecuteNuctl([]string{"deploy", functionConfig.Meta.Name, "--verbose", "--no-pull"},
+		map[string]string{
+			"path": functionPath,
+			"file": functionConfigPath,
+		})
+
+	// delete the function when we're done
+	defer suite.ExecuteNuctl([]string{"delete", "fu", functionConfig.Meta.Name}, nil)
+	suite.Require().NoError(err, "Function creation expected to pass")
+
+	// try a few times to invoke, until it succeeds
+	err = suite.ExecuteNuctlAndWait([]string{"invoke", functionConfig.Meta.Name},
+		map[string]string{
+			"method": "POST",
+			"body":   `{"return_this": "abc"}`,
+			"via":    "external-ip",
+		}, false)
+	suite.Require().NoError(err)
+
+	// get the deployed function, we're gonna inspect its resource version
+	deployedFunction, err := suite.getFunctionInFormat(functionConfig.Meta.Name, nuctl_common.OutputFormatYAML)
+	suite.Require().NoError(err)
+
+	functionResourceVersion := deployedFunction.Meta.ResourceVersion
+
+	// ensure retrieved resource version from deployed function is not wmptty
+	suite.Require().NotEmpty(functionResourceVersion)
+
+	// --- step 2 ---
+	// at this step, we're gonna change the resource version in a way it will fail duee to a conflict
+
+	// change it to anything but empty/equal to the current resource version
+	deployedFunction.Meta.ResourceVersion = functionResourceVersion[:len(functionResourceVersion)/2]
```

Comment:
I think this would be best tested with a real-world scenario instead of "faking" a resource version - save the `functionResourceVersion` in a variable, and deploy the function with some change, then use the previously saved `functionResourceVersion` which should be stale for another deployment attempt - that one should fail on resourceVersion verification.
That was you'll check it with a real-life, k8s generated, resourceVersion. And it will break if any of our assumptions would break (on a future k8s version, for example)

Reply:
Sure thing, good idea, will add